packages/lib/src/spatial-navigation/components/virtualizedGrid/helpers/convertToGrid.ts

```diff
@@ -6,10 +6,15 @@ import { NodeOrientation } from '../../../types/orientation';
 export const convertToGrid = <T extends ItemWithIndex>(
   data: T[],
   numberOfColumns: number,
+  header?: JSX.Element,
 ): GridRowType<T>[] => {
   const rows: T[][] = chunk(data, numberOfColumns);
 
-  return rows.map((items, index) => ({ items, index }));
+  return rows.map((items, index) => {
+    //We do this to have index taking into account the header
+    if (header) index++;
```

Comment:
this is mutating a parameter of the function, it's never a good idea 🙈 
I don't even understand why this works and has an effect on further iterations haha

Reply:
Done :D